### PR TITLE
Add some debug flags to internal utilities

### PR DESCRIPTION
**Pull request by BobVarioa**
Summary:
- adds the `--dont-write-results` flag to the test262 runner, which doesn't write the final results.json and just logs the differences
- adds the `--plain-results `flag to the test262 runner, which converts all the emojis into plain english 
- adds the `--debug-verbose` flag to `porf` which whenever a wasm error is thrown and the decompilation of the function is logged, it doesn't only get the surrounding lines, but the entire function instead
- adds the `--debug-surrounding` flag as an alternative to the above, instead of logging the entire function it logs the n lines around it

### CODE REVIEW

File: compiler/wrap.js [hunk 1]
```diff
@@ -188,7 +221,10 @@ export default (source, flags = [ 'module' ], customImports = {}, print = str =>
       if (!mismatch) break;
     }
 
-    if (i === wasm.length) return false;
+    if (i === wasm.length) {
+      printDecomp(-1, func, funcs, globals, exceptions)
```
Comment (CanadaHonk) [inline compiler/wrap.js:225]: nit: no `;` here and on a few other lines

File: compiler/wrap.js [hunk 2]
```diff
@@ -147,6 +147,39 @@ export default (source, flags = [ 'module' ], customImports = {}, print = str =>
...
   if (Prefs.profileCompiler) console.log(bold(`compiled in ${times[0].toFixed(2)}ms`));
 
+  const printDecomp = (middleIndex, func, funcs, globals, exceptions) => {
+    console.log(`\x1B[35m\x1B[1mporffor backtrace\u001b[0m`);
+    console.log('\x1B[4m' + func.name + '\x1B[0m');
+    
+    const surrounding = Prefs.debugSurrounding ?? 5;
+    let min = middleIndex - surrounding;
+    let max = middleIndex + surrounding + 1;
+    if (Prefs.debugVerbose) {
```
Comment (CanadaHonk) [inline compiler/wrap.js:157]: Can we also just print the entire function when `middleIndex` is ` -1`? Also could you rename this pref to something more ~obvious like `--backtrace-entire-func` or something similar? ty

**Review by CanadaHonk — CHANGES_REQUESTED** [review]
These are some nice to haves, ty! Just a few nits :)

File: compiler/wrap.js (hunk 2 above)
Comment (BobVarioa) [reply compiler/wrap.js]: Done :)

File: compiler/wrap.js (hunk 1 above)
Comment (BobVarioa) [reply compiler/wrap.js]: Fixed, at least the ones I found, pretty sure I caught them all.

**Review by CanadaHonk — APPROVED** [review]
Looks good, ty!

**Pull request merged**